arviz/data/inference_data.py

```diff
@@ -155,11 +185,12 @@ def to_netcdf(self, filename, compress=True, groups=None):
             Location of netcdf file
         """
         mode = "w"  # overwrite first, then append
-        if self._groups:  # check's whether a group is present or not.
+        if self._groups or self._groups_warmup:  # check's whether a group is present or not.
             if groups is None:
-                groups = self._groups
+                groups = self._groups + self._groups_warmup
```

Comment:
This can be made a property `self._groups + self._groups_warmup` so it doesn't need to be repeated either